### PR TITLE
Improve product list item pricing

### LINKED ISSUE #259: Improve how pricing is shown
We want to improve how prices are shown in the product list.

The spec follows from the discussion in #220 and it's basically detailed in this comment https://github.com/iFixit/react-commerce/issues/220#issuecomment-1082407628. 

- Only show a crossed-out compare_at_price if it's greater than the displayed price.
- If customer is a pro, use the normal price as the `compare_at_price` if `compare_at_price` is null in the index.
- Show the current variant result only (no range, no "from")

@sterlinghirsh @jeffsnyder anything else to add?

### PR DESCRIPTION
**Pull request by dhmacs**
fixes #259 

## Changelog

- If customer is a pro, use the normal `price` as the `compare_at_price` if `compare_at_price` is `null` in the Algolia index.

## QA

1. Visit the Vercel PR preview (replace `vercel.app` with `cominor.com`)
2. Visit the tools product list (`/Store/Tools`)
3. Verify that a pro user sees the normal price as a compare at price when a product has a discounted price for the user tier

### CODE REVIEW

File: frontend/components/product-list/sections/FilterableProductsSection/useProductSearchHitPricing.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+   compareAtPrice: number | undefined;
+   isDiscounted: boolean;
+   percentage: number;
+};
+export function useProductSearchHitPricing(
+   product: ProductSearchHit
+): ProductSearchHitPricing {
+   const user = useAuthenticatedUser();
+
+   const proTierPrice = React.useMemo(() => {
```
Comment (danielbeardsley) [inline frontend/components/product-list/sections/FilterableProductsSection/useProductSearchHitPricing.ts:17]: I'd vote for not using `useMemo()` here. I'd be willing to bet that `useMemo()` and creating this closure takes *wayy* longer than just running these if statements every time. They are just property accesses and one `parseFloat()`.

File: frontend/components/product-list/sections/FilterableProductsSection/useProductSearchHitPricing.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      }
+      return null;
+   }, [user.data?.discountTier, product.price_tiers]);
+
+   const isProUser = proTierPrice != null;
+
+   const price = proTierPrice ?? product.price_float;
+   const compareAtPrice = isProUser
+      ? product.compare_at_price ?? product.price_float
+      : product.compare_at_price;
```
Comment (danielbeardsley) [inline frontend/components/product-list/sections/FilterableProductsSection/useProductSearchHitPricing.ts:31]:
```suggestion
   const price = proTierPrice ?? product.price_float;
   const compareAtPrice = product.compare_at_price ?? product.price_float;
```

If we just always fallback to the regular price, then `isDiscounted` will still protect us from showing the same price twice.

**Comment by danielbeardsley** [comment]
CR :+1: 

Looks good @dhmacs !

deploy_block :+1:  on considering a few suggestions.

File: frontend/components/product-list/sections/FilterableProductsSection/useProductSearchHitPricing.ts (hunk 2 above)
Comment (dhmacs) [reply frontend/components/product-list/sections/FilterableProductsSection/useProductSearchHitPricing.ts]: lgtm 👍

**Comment by danielbeardsley** [comment]
CR :+1:

**Comment by jordycosta** [comment]
dev_block 🟥 

Preview 404'ing on both Parts/Tools

**Comment by dhmacs** [comment]
@jordycosta Strapi was down. I've restarted it un_dev_block 🟢

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻

**Comment by dhmacs** [comment]
un_ deploy_block 🚀

**Comment by dhmacs** [comment]
un_deploy_block 🚀

**Comment by jordycosta** [comment]
QA 🟢 
 
Discounted products now display price comparisons to the original pricing for appropriate user tiers ✔️ 
 
---
 
### Testing Procedure:
1. Navigate to the Vercel [preview](https://react-commerce-git-improve-product-list-item-pricing-ifixit.cominor.com/)
2. Click on Tools
3. Scroll down the product list until you see a discounted item
4. Ensure that the price comparison and percent off % tag is being displayed
 
> There are some discounts provided for all user tier(s), which is expected
 
---
 
### Screen Captures:
 
pricing
discount

**Pull request merged**